tools/testing/target_determination/heuristics/utils.py

```diff
@@ -20,20 +22,44 @@ def python_test_file_to_test_name(tests: Set[str]) -> Set[str]:
     return valid_tests
 
 
-def query_changed_files() -> List[str]:
-    default_branch = f"origin/{os.environ.get('GIT_DEFAULT_BRANCH', 'main')}"
+@lru_cache(maxsize=None)
+def get_pr_number() -> Optional[int]:
+    pr_number = os.environ.get("PR_NUMBER", "")
+    if pr_number == "":
+        re_match = re.match(r"^refs/tags/.*/(\d+)$", os.environ.get("GITHUB_REF", ""))
+        if re_match is not None:
+            pr_number = re_match.group(1)
+    if pr_number != "":
+        return int(pr_number)
+    return None
+
+
+@lru_cache(maxsize=None)
+def get_merge_base() -> str:
+    pr_number = get_pr_number()
+    if pr_number is not None:
+        github_token = os.environ.get("GITHUB_TOKEN")
+        headers = {
+            "Accept": "application/vnd.github.v3+json",
+            "Authorization": f"token {github_token}",
+        }
+        url = f"https://api.github.com/repos/pytorch/pytorch/pulls/{pr_number}"
```

Comment:
Are we worried about rate limiting here?

Reply:
From what I can find on rockset, we usually run ~30 pull workflows per hour, max I've seen so far is 50.  This only runs once per workflow, so probably a max of 200 per hour (I've only seen 120 max on rockset).  iirc rate limits are 5000 per person per hour so I think it'll be fine